packages/material-ui-unstyled/src/SliderUnstyled/SliderUnstyled.js

```diff
@@ -230,17 +230,19 @@ const SliderUnstyled = React.forwardRef(function SliderUnstyled(props, ref) {
   const handleChange =
     onChange &&
     ((event, value) => {
-      if (!(event instanceof Event)) event.persist();
-
       // Redefine target to allow name and value to be read.
       // This allows seamless integration with the most popular form libraries.
       // https://github.com/mui-org/material-ui/issues/13485#issuecomment-676048492
-      Object.defineProperty(event, 'target', {
+      // Clone the event to not override `target` of the original event.
+      const nativeEvent = event.nativeEvent || event;
+      const clonedEvent = new nativeEvent.constructor(nativeEvent.type, nativeEvent);
+
+      Object.defineProperty(clonedEvent, 'target', {
         writable: true,
-        value: { value, name },
+        value: { value, name, event },
```

Comment:
Why do we need to make it circular here?

Reply:
The objective was to allow developers to access the original event. I'm not aware of a use case for it, it's meant as a hedge for future cases.

Reply:
[Do developers need the event](https://github.com/mui-org/material-ui/pull/24782#discussion_r573311395) or [do they not](https://github.com/mui-org/material-ui/pull/24782#discussion_r573310380)?

Reply:
In the nominal case, we know they don't
In advanced cases, I don't know, maybe, the idea was to expose to not risk it, but we could hide it and way for a developer to complain. Happy to remove it.